src/badges.js

```diff
@@ -1275,7 +1308,10 @@ export default function insertBadges () {
   }
 
   for (const el of els) {
-    el.citeEl.insertAdjacentHTML(badgeSite.position, createBadge(el?.doi?.toLowerCase()?.trim()))
+    const doi = el.doi || (await matchReference(el.reference))?.doi
```

Comment:
It is feeling a tiny bit slow:

Would you consider:
```
  const dois = await Promise.all(els.map(el => matchReference(el.reference)))
  for (const el of els) {
    const doi = el.doi || dois[els.indexOf(el)].doi
    if (doi) {
      el.citeEl.insertAdjacentHTML(badgeSite.position, createBadge(doi.toLowerCase()?.trim()))
    }
  }
```

I think a better approach to speeding it up would be adding a batch mode to matchReference since Google Scholar is fine as its typically like 10 results but in other cases as we expand maybe some of these sites will have many more title results to match.

Reply:
I think glutton does not have a batch endpoint so better to mirror that and just send more requests probably. It is already exempt from the rate limit so could do in batches of 10 or so.

Reply:
I guess we can test both but it should be faster to use more requests as we will use more api threads